lib/renderer.js

```diff
@@ -1134,6 +1136,28 @@
         
         return hasZeroBounds(this._getContentBounds(component.document.layers));
     };
+
+    /**
+     * checks to see if there is any complex mask applied to any of the layerGroup. 
+     * Complex Mask is defined as : 
+     * (1) Bitmap masks with extendWithWhite Property (Photoshop>Layers>Layer Mask>Hide Selection)
+     * (2) Vector masks or Bitmap Mask with zero bounds.(Layer Mask>Hide All or Vector Mask>Hide All)
+     * @private
+     * @param {!Component} component 
+     * @return {boolean}
+     */
+    PixmapRenderer.prototype._componentHasComplexMasks = function (component) {
+        var layerGroupHasComplexMasks = function (layer) {
+            return layer.layers &&
+                    ((layer.mask && layer.mask.enabled && layer.mask.extendWithWhite) ||
+                    ((layer.mask || layer.path) && !layer.getTotalMaskBounds()));
```

Comment:
Style nit: can you add a space to beginning of last line to show grouping? For example:

```js
            return layer.layers &&
                   ((layer.mask && layer.mask.enabled && layer.mask.extendWithWhite) ||
                    ((layer.mask || layer.path) && !layer.getTotalMaskBounds()));
```